app/models/user.rb

```diff
@@ -2,25 +2,38 @@
 #
 # Table name: users
 #
-#  id         :integer          not null, primary key
-#  email      :string
-#  created_at :datetime         not null
-#  updated_at :datetime         not null
-#  name       :string
-#  nickname   :string
-#  image      :string
-#  uid        :integer
-#  token      :string
-#  api_token  :string
+#  id               :integer          not null, primary key
+#  email            :string
+#  created_at       :datetime         not null
+#  updated_at       :datetime         not null
+#  name             :string
+#  nickname         :string
+#  image            :string
+#  uid              :integer
+#  token            :string
+#  api_token        :string
+#  password_digest  :string
 #
 
 class User < ActiveRecord::Base
   has_many :application_users
   has_many :applications, through: :application_users
 
+  has_secure_password validations: false
+  validates_confirmation_of :password, if: :validate?
+  validates_presence_of :password, if: :validate?
```

Comment:
Prefer the new style validations `validates :column, presence: value` over `validates_presence_of`.